src/rep.rs

```diff
@@ -140,6 +146,10 @@ impl SocketRecv for RepSocket {
                     match message {
                         Message::Message(mut m) => {
                             assert!(m.len() > 1);
+                            if m.len() > 2 {
+                                self.connection_id =
```

Comment:
Taken from ZMQ spec (https://rfc.zeromq.org/spec/28/)
* An address envelope consisting of zero or more frames, each containing one identity.
For processing incoming messages:
* SHALL remove and store the address envelope, including the delimiter.

So potentially it can be more than one identity in a message. IMHO a better approach would be to store ZmqMessage in the socket - it would contain all frames up to delimiter and later we could just attach reply to it and send it down the wire